src/extensions/lodash-mixins.ts

```diff
@@ -66,5 +70,54 @@ _.mixin({
             }
         }
         return result;
-    }
+    },
+    diffObject: <BaseObjectType, ObjectToCompareType>(baseObject: BaseObjectType, objectToCompare: ObjectToCompareType): [keyof ObjectToCompareType, unknown][] => {
+        return _.differenceWith(Object.entries(objectToCompare), Object.entries(baseObject), (a, b) => {
+            return a[0] === b[0] && _.isEqual(a[1], b[1]);
+        }) as [keyof ObjectToCompareType, unknown][];
+    },
+    assignEntries: <ObjectType>(obj: ObjectType, changes: [string | number | symbol, unknown][]): ObjectType => {
+        changes.forEach(change => {
+            // For assign it becomes difficult to be type safe
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            (obj as any)[change[0]] = change[1];
+        });
+        return obj;
+    },
+    /*
+    Example
+    ```
+    const result = _.assignChanges({
+        a: 1,
+        b: 2,
+        c: 3,
+    }, {
+        a: 1,
+        b: 4,
+        d: 5
+    });
+    ```
+    Expected result
+    ```
+    a 1
+    b 4
+    c 3
+    d 5
+    ```
```

Comment:
Isn't this the same behavior as assign?
```
Object.assign({
        a: 1,
        b: 2,
        c: 3,
    }, {
        a: 1,
        b: 4,
        d: 5
    })
Object { a: 1, b: 4, c: 3, d: 5 }
```

Reply:
Nope, I talked about this in slack  bit

If I do the assign you have listed it will do
```Typescript
func (param1, param2)
param1.a = param2.a;
param1.b = param2.b;
param.d = param2.d;
```

However this function does
```Typescript
func (param1, param2)
param1.b = param2.b;
param.d = param2.d;
```
Notice that `param1.a = param2.a` is omitted because it is unchanged


This is extremely important because modifying a sequelize result is actually just an interface, the data is sitting in the underlying `dataValues` property
So if you do: `myModel.a = myModel.a`
Then `myModel.changes()` results in `['a']`

We can optimize calls by using these assign changes and only calling save if (`myModel.changes()`)